Create GitHub Actions for linting, testing and releasing

christianhueserhzdr opened the pull request:
Extend CI pipeline to do the linting and testing on each pull request and on each merging into branch main and tagging releases and to do the releasing on each tag created.

In preparation of the relocation from GitLab.com to GitHub as described in: https://gitlab.com/hifis/ansible/gitlab-role/-/issues/49

christianhueserhzdr commented:
Dear Tobias @tobiashuste, I prepared the GitHub Actions to move developments from GitLab to GitHub. Could you please have a look whether all is fine and I did not miss anything? Thanks for your review.

tobiashuste commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,87 @@
+env:
+  PY_COLORS: 1
+  ANSIBLE_FORCE_COLOR: 1
+  WORKON_HOME: ~/.cache/venv
+  PIPENV_CACHE_DIR: ~/.cache/pipenv
+  PIP_CACHE_DIR: ~/.cache/pip
```
I just learned about the caching feature of the `setup-python` action and suggest to use this instead.

```suggestion
```

tobiashuste commented on .github/workflows/prepare-action/action.yml:
```diff
@@ -0,0 +1,30 @@
+      id: setup-python
+      with:
+        python-version: '3.9'
+        
+    - name: Restore Python dependency cache.
+      uses: actions/cache@v3
+      with:
+        path: ~/.cache
+        key: ${{ runner.os }}-python-${{ steps.setup-python.outputs.python-version }}-pipenv-${{ hashFiles('Pipfile.lock') }}
+
+    - name: Install pipenv.
+      run: pip3 install --ignore-installed pipenv
+      shell: bash
```
```suggestion
```

tobiashuste commented on .github/workflows/prepare-action/action.yml:
```diff
@@ -0,0 +1,30 @@
+description: Install the necessary dependencies for jobs.
+runs:
+  using: composite
+  steps:
```
```suggestion
  steps:
    - name: Install pipenv
      run: pipx install pipenv
      shell: bash

```

tobiashuste commented on .github/workflows/prepare-action/action.yml:
```diff
@@ -0,0 +1,30 @@
+      uses: actions/setup-python@v3
+      id: setup-python
+      with:
+        python-version: '3.9'
```
According to https://github.com/actions/setup-python#caching-packages-dependencies

```suggestion
        python-version: '3.9'
        cache: 'pipenv'
```

tobiashuste reviewed (changes requested):
Thanks for implementing this! I found out about the caching feature of the `setup-python` action which is a bit simpler to apply. This is why I suggested changes to make use of it.

Could you please also add the Changelog entry?

tobiashuste commented:
@christianhueserhzdr Is this ready from your side?

christianhueserhzdr commented:
> @christianhueserhzdr Is this ready from your side?

Dear Tobias @tobiashuste, from my side I incorporated your suggestions, added the Dependabot to the GitHub Actions and removed the obsolete GitLab CI file. Could you please review again? Thanks.

tobiashuste reviewed (approved):
Thanks, @christianhueserhzdr for implementing this :shipit:
